pkg/client/example-runner.go

```diff
@@ -45,7 +46,19 @@ func RunExamplesForService(registry broker.BrokerRegistry, client *Client, servi
 				continue
 			}
 
-			if err := RunExample(client, example, service); err != nil {
+			serviceCatalogEntry, err := service.CatalogEntry()
```

Comment:
I think it would help to turn this into a function that can construct a complete list of `CompleteServiceExamples` for all the services, then another function that can filter them down to only matching ones before executing them.

Splitting it up like that would make test assertions easier into something like:

* Can we create a correct CompleteServiceExample from some fields?
* Can we create a list of those given a service?
* Can we tell if a single example matches a service and example name?

Reply:
Good idea! I refactored `RunExamplesForService` and split it up into two functions: `getAllCompleteServiceExamples`, then `runMatchingServiceExamples`. 

Please feel free to check my filtering logic in `runMatchingServiceExamples`.